.changesets/fix_strip_dashes_from_trace_id_in_customtraceidpropagator.md

```diff
@@ -0,0 +1,8 @@
+### Strip dashes from trace_id in CustomTraceIdPropagator ([Issue #4892](https://github.com/apollographql/router/issues/4892))
+
+Trace ID in opentelemetry is represented as a 128 bit number. This is usually represented as a hex string without dashes and is based on w3c trace id format.
+
+If for example, raw trace id from headers are represented by uuid4 and contains dashes, those dashes should be stripped so the raw trace id value can be parsed into a valid `trace_id`.
```

Comment:
```suggestion

The router now strips dashes from trace IDs to ensure conformance with OpenTelemetry.

In OpenTelemetry, trace IDs are 128-bit values represented as hex strings without dashes, and they're based on W3C's trace ID format.

This has been applied within the router to `trace_id` in `CustomTraceIdPropagator`.

Note, if raw trace IDs from headers are represented by uuid4 and contain dashes, the dashes should be stripped so that the raw trace ID value can be parsed into a valid `trace_id`.
```